addons/stock/wizard/stock_picking_return.py

```diff
@@ -110,20 +110,24 @@ def _prepare_move_default_values(self, return_line, new_picking):
         }
         return vals
 
+    def _prepare_picking_default_values(self):
+        return {
+            'move_lines': [],
+            'picking_type_id': self.picking_id.picking_type_id.return_picking_type_id.id or self.picking_id.picking_type_id.id,
+            'state': 'draft',
+            'origin': _("Return of %s") % self.picking_id.name,
+            'location_id': self.picking_id.location_dest_id.id,
+            'location_dest_id': self.location_id.id
+        }
+
     def _create_returns(self):
         # TODO sle: the unreserve of the next moves could be less brutal
         for return_move in self.product_return_moves.mapped('move_id'):
             return_move.move_dest_ids.filtered(lambda m: m.state not in ('done', 'cancel'))._do_unreserve()
 
         # create new picking for returned products
         picking_type_id = self.picking_id.picking_type_id.return_picking_type_id.id or self.picking_id.picking_type_id.id
```

Comment:
Because you copy it in the `_prepare_picking_default_values` you should remove it (to avoid duplicate and futur inconsistency between picking and returned values) and:
- return the picking_type of the picking instead (`picking.picking_type_id`)
- OR change the signature and only return the picking (and the `create_returns` will use the `new_picking_id.picking_type_id`)

Reply:
Good Catch,
I have returned it from new_picking